### PR TITLE
CPP: Classes and object concept c++ in video form #981

### PR DESCRIPTION
**Pull request by subhoshreep**
classes and object concept c++ video contribution
issue #981
contributed by
subhoshree pattnaik
c_cpp batch 3

<hr>

## Description 📜

classes and object concept c++ video contribution
contributed by
subhoshree pattnaik
c_cpp batch 3 
Fixes # <981> 

<hr>

## Type of change 📝

- [x] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸
image

<hr>

### CODE REVIEW

**Comment by subhoshreep** [comment]
@Suvraneel can you please review my PR.

**Comment by ChetasShree** [comment]
Amazing work @subhoshreep !!! 
@Suvraneel @Saviour1001 pls review this PR and merge it :)
Thank you

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+3.a match through application of standard c++ conversion rules.
+4. a match through application of a user defined conversion.</p>
+
+## What are the advantages of function overloading?
+<p>By overloading function we tend to save compiler from the trouble of pushing default argument values on the function call stack, hence the function does not require to test default values.
+Also a class can also have overloaded constructors. A destructor function can’t be overloaded.</p>
+
+## Example of function overloading with help of a program on calculating simple interest with default arguments.
+
+#include<iostream> <br>
```
Comment (Saviour1001) [inline C_CPP/Functions & Recursion and its types/function overloading.md:33]: You can add this code snippet via enclosing it in ```
```<Enter code here> ```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    cout<<"case2"; <br>
+    amt(1000,3); <br>
+    cout<<"case3"; <br>
+    amt(1000,3,0.11); <br>
+   cout<<"case4"; <br>
+    amt(1000,0.12); <br>
+    return 0; <br>
+} <br>
+  
+###Output of the program:
```
Comment (Saviour1001) [inline C_CPP/Functions & Recursion and its types/function overloading.md:58]: Just add a space here in the hashtags and the text

File: C_CPP/Classes & Objects Programming/classes and object video.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#### CLASSES AND OBJECT VIDEO
+LINK-https://drive.google.com/file/d/1vIXvP5JasjJ-JSZmAKNxWdgIHKKi9iDL/view?usp=sharing
```
Comment (Saviour1001) [inline C_CPP/Classes & Objects Programming/classes and object video.md:2]: You can add an image as a thumbnail and add the link to the image. [Here](https://meta.stackexchange.com/questions/38915/creating-an-image-link-in-markdown-format) is how to do that.

**Review by Saviour1001 — COMMENTED** [review]
Small changes but it will make your already awesome work a little neat. We want your work to be presented at its best !

**Comment by Suvraneel** [comment]
@subhoshreep  Why's there function overloading in a PR for classes & objects... use separate branches for separate PRs

File: C_CPP/Classes & Objects Programming/classes and object video.md (hunk 3 above)
Comment (subhoshreep) [reply C_CPP/Classes & Objects Programming/classes and object video.md]: added, can you please check @Saviour1001

**Comment by subhoshreep** [comment]
> @subhoshreep Why's there function overloading in a PR for classes & objects... use separate branches for separate PRs
> 
> I mean : Remove the unrelated file

https://github.com/girlscript/winter-of-contributing/pull/2512
@Suvraneel i have created a new pull request for it here is the link

**Comment by subhoshreep** [comment]
@Suvraneel  @Saviour1001 
I have made the required changes, can you please check once.

File: C_CPP/Functions & Recursion and its types/function overloading.md (hunk 1 above)
Comment (subhoshreep) [reply C_CPP/Functions & Recursion and its types/function overloading.md]: ok added these changes

File: C_CPP/Functions & Recursion and its types/function overloading.md (hunk 2 above)
Comment (subhoshreep) [reply C_CPP/Functions & Recursion and its types/function overloading.md]: added the changes

File: C_CPP/Classes & Objects Programming/classes and object video.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# CLASSES AND OBJECT VIDEO
+LINK-https://drive.google.com/file/d/1vIXvP5JasjJ-JSZmAKNxWdgIHKKi9iDL/view?usp=sharing 
+<br>
+![image](https://user-images.githubusercontent.com/76872340/134707897-f04f9448-e615-45cf-9cec-6c7cfb4a5f97.png)
```
Comment (vasu-1) [inline C_CPP/Classes & Objects Programming/classes and object video.md:4]:
```suggestion
# CLASSES AND OBJECT VIDEO
<br>
<iframe width="800" height="500"
src="https://drive.google.com/file/d/1vIXvP5JasjJ-JSZmAKNxWdgIHKKi9iDL/preview" 
frameborder="0" 
allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" 
allowfullscreen></iframe>
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Just Last time change this ....
by doing so your video will be embeded into .md file

Really Great Work :))
Go Ahead and do more contribution !!

File: C_CPP/Classes & Objects Programming/classes and object video.md (hunk 4 above)
Comment (subhoshreep) [reply C_CPP/Classes & Objects Programming/classes and object video.md:4]: image
this is showing as preview

**Comment by Suvraneel** [comment]
@subhoshreep  🚀

**Pull request merged**